src/editor/Gui.cpp

```diff
@@ -2607,20 +2946,183 @@ void Gui::drawAbout() {
+
+					if (Optimize) {
+						logf("    Optmizing...\n");
+						maps[i]->delete_unused_hulls().print_delete_stats(2);
+					}
+
+					logf("\n");
+				}
+
+				BspMerger merger;
+				Bsp* result = merger.merge(maps, vec3(), Path, NoRipent, NoScript);
+
+				logf("\n");
+				if (result->isValid()) result->write(Path);
+				logf("\n");
+				result->print_info(false, 0, 0);
+
+				g_app->clearMaps();
+
+				fixupPath(Path, FIXUPPATH_SLASH::FIXUPPATH_SLASH_SKIP, FIXUPPATH_SLASH::FIXUPPATH_SLASH_SKIP);
+
+				if (fileExists(Path))
+				{
+					result = new Bsp(Path);
+					g_app->addMap(result);
+				}
+				else
+				{
+					logf("Error while map merge!\n");
+					g_app->addMap(new Bsp());
+				}
+			}
+			showMergeMapWidget = false;
+		}
+	}
+
+	ImGui::End();
+}
+
+void Gui::drawImportMapWidget() {
+	ImGui::SetNextWindowSize(ImVec2(500, 140), ImGuiCond_FirstUseEver);
+	ImGui::SetNextWindowSizeConstraints(ImVec2(500, 140), ImVec2(500, 140));
+	static char Path[256];
+	const char* title = "Import .bsp model as func_breakable entity";
+
+	if (showImportMapWidget_Type == SHOW_IMPORT_OPEN)
+	{
+		title = "Open map";
+	}
+	else if (showImportMapWidget_Type == SHOW_IMPORT_ADD_NEW)
+	{
+		title = "Add map to renderer";
+	}
+
+	if (ImGui::Begin(title, &showImportMapWidget)) {
+		ImGui::InputText(".bsp file", Path, 256);
+		if (ImGui::Button("Load", ImVec2(120, 0)))
+		{
+			fixupPath(Path, FIXUPPATH_SLASH::FIXUPPATH_SLASH_SKIP, FIXUPPATH_SLASH::FIXUPPATH_SLASH_SKIP);
+			if (fileExists(Path))
+			{
+				logf("Loading new map file from %s path.\n", Path);
+				showImportMapWidget = false;
+				if (showImportMapWidget_Type == SHOW_IMPORT_ADD_NEW)
+				{
+					g_app->addMap(new Bsp(Path));
+				}
+				else if (showImportMapWidget_Type == SHOW_IMPORT_OPEN)
+				{
+					g_app->clearMaps();
+					g_app->addMap(new Bsp(Path));
+				}
+				else
+				{
+					if (g_app->mapRenderers.size() && g_app->mapRenderers[0]->map)
+					{
+						Bsp* model = new Bsp(Path);
+						if (!model->ents.size())
+						{
+							logf("Error! No worldspawn found!\n");
+							delete model;
+						}
+						else
+						{
+							Bsp* map = g_app->getSelectedMap()->map;
+							logf("Binding .bsp model to func_breakable.\n");
+							Entity* tmpEnt = new Entity("func_breakable");
+							tmpEnt->setOrAddKeyvalue("origin", model->ents[0]->getOrigin().toKeyvalueString());
+							tmpEnt->setOrAddKeyvalue("gibmodel", "models/" + basename(Path));
+							tmpEnt->setOrAddKeyvalue("model", "models/" + basename(Path));
+							tmpEnt->setOrAddKeyvalue("spawnflags", "1");
+							map->ents.push_back(tmpEnt);
+							map->update_ent_lump();
+							logf("Success! Now you needs to copy model to path: %s\n", ("models/" + basename(Path)).c_str());
+
+							if (app->pickInfo.map->GetBspRender())
+							{
+								app->pickInfo.map->GetBspRender()->preRenderEnts();
+								app->updateEntConnections();
+								app->updateEntConnectionPositions();
+							}
+							
+							app->reloadBspModels();

```

Comment:
this doesn't seem to do anything besides add a func_breakable object with a model key set to a bsp file that doesn't exist. I think it's better to just document somewhere that it's possible to use BSPs as entity models.